snowflake/data_access.go

```diff
@@ -221,6 +229,118 @@ func (s *AccessSyncer) SyncAccessProviderMasksToTarget(ctx context.Context, apTo
 	return nil
 }
 
+func (s *AccessSyncer) SyncAccessProviderFiltersToTarget(ctx context.Context, apToRemoveMap map[string]*importer.AccessProvider, apMap map[string]*importer.AccessProvider, roleNameMap map[string]string, feedbackHandler wrappers.AccessProviderFeedbackHandler, configMap *config.ConfigMap) error {
+	if configMap.GetBoolWithDefault(SfStandardEdition, false) {
+		if len(apToRemoveMap) > 0 || len(apMap) > 0 {
+			logger.Error("Skipping filter policies due to Snowflake Standard Edition.")
+		}
+
+		return nil
+	}
+
+	logger.Info(fmt.Sprintf("Configuring access provider as filters in Snowflake. Update %d masks remove %d masks", len(apMap), len(apToRemoveMap)))
+
+	repo, err := s.repoProvider(configMap.Parameters, "")
+	if err != nil {
+		return err
+	}
+
+	//Groups filters by table
+	updateGroupedFilters, err := groupFiltersByTable(apMap, feedbackHandler)
+	if err != nil {
+		return err
+	}
+
+	removeGroupedFilters, err := groupFiltersByTable(apToRemoveMap, feedbackHandler)
+	if err != nil {
+		return err
+	}
+
+	feedbackFn := func(aps []*importer.AccessProvider, actualName *string, externalId *string, err error) error {
+		var feedbackErr error
+
+		var errorMessages []string
+
+		if err != nil {
+			errorMessages = []string{err.Error()}
+		}
+
+		for _, ap := range aps {
+			var actualNameStr string
+			if actualName != nil {
+				actualNameStr = *actualName
+			} else if ap.ActualName != nil {
+				actualNameStr = *ap.ActualName
+			}
+
+			var apExternalId *string
+			if externalId != nil {
+				apExternalId = externalId
+			} else {
+				apExternalId = ap.ExternalId
+			}
+
+			ferr := feedbackHandler.AddAccessProviderFeedback(importer.AccessProviderSyncFeedback{
+				AccessProvider: ap.Id,
+				ActualName:     actualNameStr,
+				ExternalId:     apExternalId,
+				Errors:         errorMessages,
+			})
+			if ferr != nil {
+				feedbackErr = multierror.Append(feedbackErr, ferr)
+			}
+		}
+
+		return feedbackErr
+	}
+
+	//Create or update filters per table
+	updatedTables := set.NewSet[string]()
+
+	for table, filters := range updateGroupedFilters {
+		filterName, externalId, createErr := s.updateOrCreateFilter(ctx, repo, table, filters, roleNameMap)
+
+		// TODO remove old versions of filters
```

Comment:
Do you have a separate issue for this?

Reply:
Old comment of during development. The old versions of the fitlers are removed in the repository (repository.go line 759)